### ISSUE TITLE
build error:error: command '/usr/local/cuda/bin/nvcc' failed with exit status 1

### ISSUE THREAD

**Issue by hubert0527** [NVIDIAGameWorks/kaolin#4]
Repost (seems like the project is reset)

I  faced this error with `pytorch 1.2.0`, `cuda 10.0` and `gcc 5.4.0`.
If you face the same error, try to modify `#include <torch/extension.h>` to `#include <torch/types.h>` in the following files (`*.h` and `*.cu` files only):
```
kaolin/cuda/util.h
kaolin/cuda/cuda_util.h
kaolin/cuda/mesh_intersection_cuda.cu
kaolin/cuda/sided_distance_cuda.cu
```

**Comment by sailor-z** [comment]
I have modified the files listed above, but it still has the same error.

**Comment by sailor-z** [comment]
All the errors occur in this file:
![image](https://user-images.githubusercontent.com/26895204/68957386-eadc8c80-0797-11ea-938b-e4adeb54400f.png)

**Comment by hubert0527** [comment]
@sailor-z  Could you show a more complete screenshot of the errors?

**Comment by sailor-z** [comment]
![image](https://user-images.githubusercontent.com/26895204/68960163-9e944b00-079d-11ea-8177-919d92916bdd.png)
@hubert0527

**Comment by MtlsBart** [comment]
After this and some other local fixes (including setting the return type of MeshIntersectionKernelLauncher the same in both mesh_intersection_cuda.cu and mesh_intersection.cpp) , I keep failing to get a working solution. Currently: `ModuleNotFoundError: No module named 'kaolin.cuda.sided_distance'` . **Does anyone have a working Windows solution, and if so, which version of MSVC/PyTorch/Cuda did you use ?**

**Comment by arozantsev** [comment]
Hi @MtlsBart,

I have tried two cases:
1. CUDA 10.0, V10.0.130; PyTorch 1.2.0; MSVC 14.16.27023 
2. CUDA 10.1, V10.1.243; PyTorch 1.3.1; Don't remember the MSVC version right now -- I will post in the comment later

and in both cases it worked after the following two fixes:
1. There is one known PyTorch issue that prevents from installation:
   * The error is : C:/Python37/lib/site-packages/torch/include\torch/csrc/jit/argument_spec.h(190): error: member "torch::jit::ArgumentSpecCreator::DEPTH_LIMIT" may not be initialized
   * To fix this issue - follow: https://github.com/facebookresearch/detectron2/issues/9#issuecomment-549013090
2. There is another cuda related bug, to fix which the following needs to be done:
   * In file: kaolin\cuda\mesh_intersection.cpp - change line 20: from
``` int MeshIntersectionKernelLauncher(```
to
```void MeshIntersectionKernelLauncher(```
    * This bug will be addressed in future commits

``Note``: the following files should **not** be modified.
```
kaolin/cuda/util.h
kaolin/cuda/cuda_util.h
kaolin/cuda/mesh_intersection_cuda.cu
kaolin/cuda/sided_distance_cuda.cu
```

**Comment by radekd91** [comment]
@arozantsev , thanks a lot for your hints. It helped me set it up but I had to add one more change, because cuda was failing compilation of files because of the following error: 

`error: function “atomicAdd(double *, double)” has already been defined`

I had to look up `atomicAdd` in the entire kaolin project and it turns out it's defined in a few files and it is always inclosed in an #if-#endif which looks like this: 
`#if  __CUDA_ARCH__ < 600 and defined(__CUDA_ARCH__)` 
This is wrong because `and` is not a valid opertator and should be replaced with `&&` to look like this: 
`#if  __CUDA_ARCH__ < 600 && defined(__CUDA_ARCH__)`
After fixing this in:  
`soft_rasterize_cuda_kernel.cu`
`rasterize_cuda_kernel.cu`
`voxelization_cuda_kernel.cu`
`rasterizer_cuda_kernel.cu`
the compilation finished just fine. Note some of these filenames occur multiple times in different folders. Fix all of them.

Then I had another problem running `python setup.py install`. Inside this script, pip is trying to install `shapely` which failed for me. I had to go in and install it separately through conda. 
`conda install -c conda-forge shapely `

Then I was able to rerun `python setup.py install` and import kaolin successfully. 

Hope this helps.
